concepts/05 UI Components/DataGrid/15 Columns/40 Customize Cells/2 Customize Cell Appearance.md

```diff
@@ -0,0 +1,253 @@
+            }
+        });
+    });
+
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-data-grid
+        (onCellPrepared)="onCellPrepared($event)">
+    </dx-data-grid>
+
+    <!-- tab: app.component.ts -->
+    // ...
+    export class AppComponent {
+        onCellPrepared (e) {
+            if (e.rowType === "data") {
+                if (e.column.dataField === "Speed" && e.data.Speed > e.data.SpeedLimit) {
+                    e.cellElement.style.cssText = "color: white; background-color: red";
+                    // or
+                    e.cellElement.classList.add("my-class");
+                }
+            }
+        }
+    }
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid ...
+            @cell-prepared="onCellPrepared">
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DxDataGrid from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid
+        },
+        methods: {
+            onCellPrepared(e) {
+                if (e.rowType === "data") {
+                    if (e.column.dataField === "Speed" && e.data.Speed > e.data.SpeedLimit) {
+                        e.cellElement.style.cssText = "color: white; background-color: red";
+                        // or
+                        e.cellElement.classList.add("my-class");
+                    }
+                }
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DataGrid from 'devextreme-react/data-grid';
+
+    const cellPrepared = (e) => {
+        if (e.rowType === "data") {
+            if (e.column.dataField === "Speed" && e.data.Speed > e.data.SpeedLimit) {
+                e.cellElement.style.cssText = "color: white; background-color: red";
+                // or
+                e.cellElement.classList.add("my-class");
+            }
+        }
+    }
+
+    function App() {
+        return (
+            <DataGrid ...
+                onCellPrepared={cellPrepared}>
+            </DataGrid>
+        );
+    }
+
+    export default App;
+    
+---
+
+You can also specify the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate') property. It allows you to replace cell content with custom markup, but does not give you access to the entire cell element.
+
+The example below shows how to use the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate') property to implement the same task. 
```

Comment:
`to implement the same task` I'm not sure you can implement a task - a function or feature or functionality, yes